hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieFileReader.java

```diff
@@ -34,7 +35,17 @@
 
   public Set<String> filterRowKeys(Set<String> candidateRowKeys);
 
-  public Iterator<R> getRecordIterator(Schema schema) throws IOException;
+  public Iterator<R> getRecordIterator(Schema readerSchema) throws IOException;
+
+  default Iterator<R> getRecordIterator() throws IOException {
+    return getRecordIterator(getSchema());
+  }
+
+  public Option<R> getRecordByKey(String key, Schema readerSchema) throws IOException;
```

Comment:
We can introduce that config but how will it be used? 

I feel the lookup by key is only useful for internal features (like RFC-15 or RFC-08) rather than a generic API for HUDI. HUDI record keys tend to be UUDIs which are large and looking them up is not a common usecase.

Reply:
I think. throwing `UnSupportedException` is actually better. nothing can fundamentally change whether a format supports this or not